realm/realm-library/src/main/cpp/util.hpp

```diff
@@ -572,6 +574,10 @@ class JniByteArray {
         , m_arrayLength(javaArray == NULL ? 0 : env->GetArrayLength(javaArray))
         , m_array(javaArray == NULL ? NULL : env->GetByteArrayElements(javaArray, NULL))
         , m_releaseMode(JNI_ABORT) {
+        if (m_javaArray != nullptr && m_array == nullptr) {
+            // javaArray is not null but GetByteArrayElements returns null, something is really wrong.
```

Comment:
It is `null` only when some serious thing happened in the jvm when calling `GetByteArrayElements`, merely our fault or user's fault. Basically it is something should never happen.